File: ts/easy/isPalindrome.ts
export const isPalindrome = (str: string) => {
  const reversed = str.toLowerCase().split('').reverse().join('');
  return str.toLowerCase() === reversed;
};
console.log(isPalindrome('Ana')); //	true
console.log(isPalindrome('cat')); //	false
// O(n)

// bad solution: too much recursion
const isPalindromeR = (str: string): boolean => {
  if (str.length <= 1) {
      return true;
  }
  if (str[0] !== str[str.length - 1]) {
      return false;
  }
  str.slice(1,-1);
  return isPalindromeR(str);
}
// console.log(isPalindromeR('a'));
// console.log(isPalindromeR('ana'));
// console.log(isPalindromeR('racecar'));
// console.log(isPalindromeR('car'));

File: python/easy/is_palindrome.py
def reverseStr(str):
  counter = 0;
  reversedStr = ''
  
  while(counter < len(str)):
    counter += 1
    reversedStr += str[len(str) - counter]
  return reversedStr
  
def is_palindrome(str):
	reversedStr = reverseStr(str);
	return str.lower() == reversedStr.lower()

print(is_palindrome('Ana'));	# True
print(is_palindrome('cat'));	# False
#	O(n)

def is_palindrome_r(s):
    if len(s) <= 1:
        return True
    if s[0] != s[len(s) - 1]:
        return False
    s = s[1:-1]
    return is_palindrome_r(s)
print(is_palindrome_r("racecar"))
